📖 [bento] Add documentation for social-share, date-countdown, accordion: 1.0 (#31920)
ampproject/amphtml: extensions/amp-accordion/amp-accordion.md

The example below contains an `amp-accordion` with three sections. The
`expanded` attribute on the third section expands it on page load.

> [example preview="top-frame" playground="true" imports="amp-accordion:1.0"]

Reviewer: @sebastianbenz will this syntax work? Should we include it in the documentation? https://amp.dev/documentation/guides-and-tutorials/contribute/contribute-documentation/formatting/?format=websites#preview-code-samples

[/example]

### Migrating from 0.1

> Unlike `0.1`, the experimental `1.0` version of `amp-accordion` does not currently support session states. It behaves as if the `disable-session-states` attribute is always set.
> 
> Both `0.1` and `1.0` versions of `amp-accordion` can be used with `amp-bind`. However, in `1.0` `amp-bind` can bind directly to the `expanded` attribute whereas in `0.1` `amp-bind` operates on the `[data-expand]` attribute. The `[data-expand]` attribute is no longer used in `1.0`. See the `expanded` attribute in the next section for more information.

Reviewer: ```suggestion
The experimental `1.0` version of `amp-accordion` does not support session states. It behaves as if the `disable-session-states` attribute is always applied.

Version `0.1` and `1.0` are compatible with `amp-bind`, but some binding syntax is different. You may bind directly with the `expanded` attribute in version `1.0`. The `[data-expanded]` is not supported in version `1.0`. See the `expanded` attribute below for further information.
```

Author: Thanks, I used this suggestion in the latest revision.

[/example]

### expanded

Apply the `expanded` attribute to a nested `<section>` to expand that section when the page loads.

> Bind the `expanded` attribute to a `<section>` to expand or collapse that

Reviewer 2: ```suggestion
Bind the `[expanded]` attribute to a `<section>` to expand or collapse that
```

Author: Thanks, I used this suggestion in the latest revision.

> section. An expanded section collapses if the expression evaluates as false. A

Reviewer: Is there a syntax to just toggle the expand/collapse, or do we have to specify as true or false?

Author: Unfortunately not for the bind examples.

> Bind the `expanded` attribute to a `<section>` to expand or collapse that
> section. An expanded section collapses if the expression evaluates as false. A
> collapsed section expands if the expression evaluates as anything other than
> false.

Reviewer: ```suggestion
Use `amp-bind` to bind the `expanded` attribute to programmatically expand or collapse a `<section>` element. An expanded section collapses when the expression evaluates as `false`. A collapsed section expands if the expression evaluates to anything other than `false`.
```

Author: Thanks, I used this suggestion in the latest revision.

> false.

Reviewer 2: Is this also the case for falsy values?

Author: Not the case for falsy values interestingly enough.

[example preview="top-frame" playground="true" imports="amp-accordion:1.0"]

```html
<amp-accordion>
  <section
    [expanded]="sectionOne"

>     on="expand:AMP.setState({sectionOne: true});collapse:AMP.setState({sectionOne: false})"

Reviewer 3: /cc @caroqliu @sebastianbenz @CrystalOnScript  This is a good example of Bento-vs-AMP world. In Bento we recommend to call API, in AMP this is the right path. Not sure yet how to separate these.

Reviewer 4: For now, I think it's best to have a separate example using the JS API in the bento section of the docs.

Author: To clarify, I should create another example in this doc?  Using a call to API?

Reviewer 3: Yep. I'd have two sub-sections: AMP example and Bento example to call imperative APIs.

Reviewer 3: The Bento example could be much smaller. E.g. "In Bento mode the same APIs can be called via `getApi()`, for instance:"

Author: Great!  This has been added here (at the end of the `Actions` section in the accordion documentation): https://github.com/ampproject/amphtml/pull/31920/files#diff-ab5b7d1f812d4c22e6a887e4324be32ec144b20fcf7075b971d968f65616dea5R260

Author: I've created a separate PR for further discussion on this!
https://github.com/ampproject/amphtml/pull/32052

@sebastianbenz @CrystalOnScript @dvoytenko

  >
    <h2>Section 1</h2>
    <p>Bunch of awesome content</p>
  </section>
  <section>
    <h2>Section 2</h2>
    <div>Bunch of awesome content</div>
  </section>
  <section expanded>
    <h2>Section 3</h2>
    <div>Bunch of awesome content</div>
  </section>
</amp-accordion>
<button on="tap:AMP.setState({sectionOne: true})">Expand section 1</button>
<button on="tap:AMP.setState({sectionOne: false})">Collapse section 1</button>
```

[/example]

### expand-single-section

> Apply the `expand-single-section` attribute to `amp-accordion` to specify that
> only one `<section>` can expand at a time. If the user clicks or taps on a
> collapsed `<section>`, any currently expanded `<section>` collapses.

Reviewer: ```suggestion
Allow only one section to expand at a time by applying the `expand-single-section` attribute to the `<amp-accordion>` element. This means if a user taps on a collapsed `<section>`, it will expand and close other open `<section>`'s.
```

Author: Thanks, I used something similar in the latest revision.  Replaced "close" and "open" with "collapsed" and "expanded" in the last sentence.
